README.md

```diff
@@ -2,24 +2,134 @@
+
 ## Installation
 
-:construction: WIP
+We will soon make a first release of the library on PyPi. In the meantime, you can clone the repository locally and install directly from source:
+
+```shell
+git clone https://github.com/dynamiqs/dynamiqs.git
+pip install -e dynamiqs
+```
+
+## Examples
+
+### Simulate a leaky bosonic mode
+
+Let's simulate a leaky cavity with Hamiltonian $H=\Delta a^\dagger a$ and a single Lindblad jump operator $L=\sqrt{\kappa} a$, using QuTiP to define these operators:
+
+```python
+import dynamiqs as dq
+import numpy as np
+import qutip as qt
+import torch
+
+# parameters
+n = 128
+delta = 1.0
+kappa = 0.5
+alpha0 = 1.0
+
+# QuTiP operators, initial state and saving times
+a = qt.destroy(n)
+H = a.dag() * a
+jump_ops = [np.sqrt(kappa) * a]
+rho0 = qt.coherent(n, alpha0)
+t_save = np.linspace(0, 1.0, 101)
+
+# uncomment the next line to run the simulation on GPU
+# torch.set_default_device('gpu')
 
-## Usage
+# simulation
+result = dq.mesolve(H, jump_ops, rho0, t_save)
+print(result)
+```
 
-:construction: WIP
+Output:
+
+```shell
+|██████████| 100.0% - time 00:00/00:00
+==== Result ====
+Method       : Dopri5
+Start        : 2023-07-06 22:52:30
+End          : 2023-07-06 22:52:31
+Total time   : 0.61 s
+states       : Tensor (101, 128, 128) | 12.62 Mb
+```
+
+### Compute the gradient with respect to some parameters
+
+Suppose that in the above example, we want to compute the gradient of the final state photon number expectation value $\text{loss}=\braket{a^\dagger a}=\mathrm{Tr}[a^\dagger a \rho]$ with respect to the decay rate $\kappa$ and the initial coherent state amplitude $\alpha_0$. For this computation, we will define the objects with dynamiqs:
+
+```python
+import dynamiqs as dq
+import numpy as np
+import torch
+
+# parameters
+n = 128
+delta = 1.0
+kappa = torch.tensor([0.5], requires_grad=True)
+alpha0 = torch.tensor([1.0], requires_grad=True)
+
+# dynamiqs operators, initial state and saving times
+a = dq.destroy(n)
+H = a.mH @ a
+jump_ops = [torch.sqrt(kappa) * a]
+rho0 = dq.coherent(n, alpha0)
+t_save = np.linspace(0, 1.0, 101)
+
+# uncomment the next line to run the simulation on GPU
+# torch.set_default_device('gpu')
+
+# simulation
+options = dq.options.Dopri5(gradient='autograd', verbose=False)
+result = dq.mesolve(H, jump_ops, rho0, t_save, options=options)
+
+# gradient computation
+loss = dq.expect(a.mH @ a, result.states[-1])  # Tr[a^dag a rho]
+grads = torch.autograd.grad(loss, (kappa, alpha0))
+print(grads)
+```
+
+Output:
+
+```shell
+(tensor([-0.6065]), tensor([1.2131]))
+```
 
 ## Performance
 
-:construction: WIP
+While we have not yet conducted a rigorous benchmark of the library, we have some preliminary results. For instance, simulating two coupled dissipative bosonic modes each truncated at 32 photons takes approximately **2 minutes** with QuTiP, compared to **4 seconds** with dynamiqs (when executed on a standard GPU).
```

Comment:
This comment is not super rigourous, I would keep it out for now. Once we have clearer benchmarks, we can add something similar back.